pymeasure/instruments/tdk/tdk_base.py

```diff
@@ -0,0 +1,387 @@
+
+    Do not directly instantiate an object with this class. Use one of the
+    TDK-Lambda power supply instrument classes that inherit from this parent
+    class.
+
+    Untested commands are noted in docstrings.
+    """
+
+    # ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+    # Initializer and important communication methods
+    # ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+    def __init__(self, adapter, name="TDK-Lambda Base", address=6, **kwargs):
+        """
+        The initialization of a TDK instrument requires the current address
+        of the TDK power supply. The default address for the TDK Lambda is 6.
+
+        :param adapter: VISAAdapter instance
+        :param name: Instrument name
+        :param address: Serial port daisy chain number. Default is 6.
+        """
+        super().__init__(
+            adapter,
+            name,
+            includeSCPI=False,
+            asrl={'read_termination': "\r", 'write_termination': "\r"},
+            **kwargs
+        )
+        self.address = address
+
+    def write(self, command):
+        """Replace the ``Instrument.write()`` method to strip out an "OK" reply
+        that the instrument returns for any non-querying commands.
+
+        By default, any non-querying commands (i.e., a command that does NOT
+        have the "?" symbol in it like the instrument command "PV 10") will
+        automatically return an "OK" reply. This is done to confirm that the
+        instrument has received the command. Any querying commands (i.e., a
+        command that does have the "?" symbol in it like the instrument command
+        "PV?") will return the requested value. The returned value itself is
+        confirmation that the command was received.
+
+        The default, the ``Instrument.write()`` method is not set up to
+        automatically strip out instrument confirmations. If the command
+        string does not contain a "?", then ``self.read()`` once to clear the
+        "OK" from the read buffer. If this is not done, the adapter read buffer
+        will hold the "OK" reply until the next read command is given.
+
+        :param command: Command string to be sent to the instrument.
+        """
+
+        super().write(command)
+        if '?' not in command:
+            # clear "OK" from the adapter read buffer
+            self.read()
+
+    # ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+    # Properties
+    # ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+    address = Instrument.setting(
+        "ADR %d",
+        """Set the address of the power supply.
+
+        Valid values are integers between 0 - 30 (inclusive).""",
+        validator=strict_discrete_set,
+        values=range(0, 31)
+    )
+
+    remote = Instrument.control(
+        "RMT?", "RMT %s",
+        """Get the current remote operation of the power supply.
```

Comment:
It is a control, so please use "Control" as the first word!

Reply:
Updated in last commit